Add the ability to customize curl options.

ahpnils opened the pull request:
Customizing curl options without editing the
script can be useful in some cases, depending on
the used vault setup. The new option set allows
a variety of setup, like redirects and
self-signed certificates, and we can remove the
silent option for debug purposes.

Commit:
Add the ability to customize curl options.

Customizing curl options without editing the
script can be useful in some cases, depending on
the used vault setup. The new option set allows
a variety of setup, like redirects and
self-signed certificates, and we can remove the
silent option for debug purposes.

ttalle commented on vault.example.inc:
```diff
@@ -3,3 +3,5 @@ VAULT_SECRET_ID="cccccccc-dddd-eeee-ffff-ggggggggggggg"
 VAULT_ADDRESS="https://127.0.0.1:8200"
 VAULT_SECRET_BASE="secret/ca"
+
+CURL_OPTS=(--silent --insecure --location)
```
Hey Nils, thank you for your contribution! Adding separate Curl options is a good idea, but adding --insecure as default in the example config seems like a bad precedent. Could you change that?

ahpnils replied:
Hello, sorry about the insecure option. My latest commit should fix this.

ttalle replied:
Great, thanks!

ahpnils commented:
Is there anything I should add, remove or fix ?

ttalle commented:
> Is there anything I should add, remove or fix ?

Hey Nils, I'm not much of a Github user so I submitted a change request but apparently it was stuck in _Pending_ state for six months (and obviously not visible for anyone else).

I've posted the change request. I think the default should be secure so if you can remove the --insecure from the default options I'm happy to merge this pull-request.

ahpnils committed:
remove insecure option

removed curl  --insecure argument following https://github.com/ttalle/dehydrated-vault-hook/pull/2#issuecomment-700269602
